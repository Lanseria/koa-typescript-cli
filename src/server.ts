import * as Koa from "koa";
import * as Router from "koa-router";
import * as logger from "koa-logger";
import * as json from "koa-json";
import * as bodyParser from "koa-bodyparser";

const app = new Koa();
const router = new Router();

interface HelloRequest {
  name: String;
}

router.post("/", async (ctx, next) => {
  const data = <HelloRequest>ctx.request.body;
  ctx.body = { name: data.name };
  await next();
});

app.use(json());
app.use(logger());
app.use(bodyParser());

app.use(router.routes()).use(router.allowedMethods());

app.listen(3000, () => {
  console.log("Server running on port http://localhost:3000");
});
