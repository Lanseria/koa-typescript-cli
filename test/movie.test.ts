import request from "supertest";
import { eApp } from "../src/server";
import Result from "../src/vo/result.vo";
import Mock from "mockjs";

const movieUrlPrefix = "/movies";

test(`[POST] ${movieUrlPrefix} add a random movie`, async () => {
  const app = await eApp;
  const randomObj = {
    name: Mock.mock("@name"),
    releaseYear: Mock.mock('@date("yyyy")'),
    rating: Mock.mock("@integer(60, 100)")
  };
  const response: request.Response = await request(app.callback())
    .post(`${movieUrlPrefix}`)
    .send(randomObj);
  expect(response.status).toBe(200);
  const result: Result<typeof randomObj> = JSON.parse(response.text);
  expect(result.data.name).toBe(randomObj.name);
  expect(result.data.releaseYear).toBe(randomObj.releaseYear);
  expect(result.data.rating).toBe(randomObj.rating);
});
