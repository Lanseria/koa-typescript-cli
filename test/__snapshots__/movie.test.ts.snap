// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[POST] /movies add a random movie 1`] = `"{\\"code\\":0,\\"msg\\":\\"success\\",\\"data\\":{\\"name\\":\\"helol\\",\\"releaseYear\\":133,\\"rating\\":132,\\"id\\":\\"3cdf7b04-f788-4729-b9a4-91f488b1f67d\\"}}"`;
